server/models/questions/getReaderQuestions.js

```diff
@@ -0,0 +1,135 @@
+const { pool } = require('../database')
+
+/**
+ *
+ * @param {Array} sortParams
+ * @return {String}
+ */
+function sortQuery(sortParams) {
+  if (!Array.isArray(sortParams)) {
+    return ''
+  }
+  return (
+    ' ORDER BY ' +
+    sortParams
+      .map(({ id, desc }) => `${id} ${desc ? 'DESC' : 'ASC'}`)
+      .join(', ')
+  )
+}
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Number} param0.search
+ * @param {Number} param0.tag_ids
+ * @param {String} param0.difficulty
+ * @param {Number} param0.per_page_size
+ * @param {Number} param0.page
+ * @param {Array} param0.sort
+ * @return {Promise}
+ */
+function getReaderQuestions({
+  username,
+  search,
+  tag_ids: tagIds,
+  difficulty,
+  per_page_size: perPageSize,
+  page,
+  sort,
+}) {
+  return new Promise((resolve, reject) => {
+    let countQuery = `SELECT COUNT(DISTINCT q.id) AS total FROM questions q
+                      INNER JOIN questions_editors qe ON q.id=qe.question_id `
+    const cQArr = []
+
+    let query = `SELECT DISTINCT q.id, q.name, q.difficulty, q.creator FROM questions q
+                INNER JOIN questions_editors qe ON q.id=qe.question_id `
+    const qArr = []
+
+    if (
+      tagIds &&
+      Array.isArray(tagIds) &&
+      !tagIds.some(isNaN) &&
+      tagIds.length
+    ) {
+      countQuery += `INNER JOIN questions_tags qt ON q.id=qt.question_id WHERE qt.tag_id IN(`
+      query += `INNER JOIN questions_tags qt ON q.id=qt.question_id WHERE qt.tag_id IN(`
+
+      tagIds.forEach((tagId, idx) => {
+        query += `?`
+        countQuery += `?`
+        if (idx != tagIds.length - 1) {
+          query += `,`
+          countQuery += `,`
+        } else {
+          query += `)`
+          countQuery += `)`
+        }
+        qArr.push(+tagId)
+        cQArr.push(+tagId)
+      })
+      query += ` AND `
+      countQuery += ` AND `
+    } else {
+      query += `WHERE `
+      countQuery += `WHERE `
+    }
+
+    query += ` qe.editor=? `
+    countQuery += ` qe.editor=? `
+    qArr.push(username)
+    cQArr.push(username)
```

Comment:
#### NEW

1. same as getEditorQuestions, only no need of providing `access` here